lib/rubocop/options.rb

```diff
@@ -23,8 +23,14 @@ def parse(command_line_args)
 
       @validator.validate_compatibility
 
-      if @options[:stdin] && !args.one?
-        raise ArgumentError, '-s/--stdin requires exactly one path.'
+      if @options[:stdin]
+        if args.any?
```

Comment:
In the end I couldn't come up with a message that better than the one we've got, so I only updated the comments.